### PR TITLE
Preserve python2 on Debian 9

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request:
* Pulls in upstream changes
* Changes the Ansible role so that nothing is done on Debian 9.  In other words, python2 is preserved on Debian 9.

## 💭 Motivation and Context

In cisagov/ansible-role-bloodhound#7 I added to the molecule prepare playbook the update and python playbooks that we normally run via packer before building up a COOL AMI.  (See [here](https://github.com/cisagov/guacamole-packer/blob/develop/src/packer.json#L65-L79), for example.)  This gives molecule a more realistic base from which to start applying Ansible roles, so I also want to backport these additions to [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).

@mcdonnnj is fine with adding the upgrade and python playbooks to the CyHy packer code, and he is also fine with adding them to the CyHy Ansible roles _as they are upgraded to python3_.  Since the CyHy AMIs and Ansible roles will drop support for Debian 9 and move to Debian 10 as part of the python3 upgrade, having this Ansible role preserve python2 on Debian 9 will stop our lineage and apb tools from forcing his hand before he is ready.  That is the context and motivation for these changes.

## 🧪 Testing

All pre-commit hooks pass.  I modified the molecule tests to check that Python2 is preserved on Debian 9, then verified that the modified tests pass after these changes.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,9 @@
 [![Total alerts](https://img.shields.io/lgtm/alerts/g/cisagov/ansible-role-remove-python2.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/cisagov/ansible-role-remove-python2/alerts/)
 [![Language grade: Python](https://img.shields.io/lgtm/grade/python/g/cisagov/ansible-role-remove-python2.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/cisagov/ansible-role-remove-python2/context:python)
 
-An Ansible role for removing all python2 packages.
+An Ansible role for removing all python2 packages on all distributions
```
Comment (mcdonnnj) [inline README.md]: Nitpick: "python2" should be capitalized.

File: molecule/default/tests/test_default.py [hunk 2]
```diff
@@ -16,5 +16,8 @@
     "f", ["/usr/bin/python", "/usr/bin/python2", "/usr/bin/python2.7"]
 )
 def test_python(host, f):
-    """Ensure that python2-specific files no longer exist."""
-    assert not host.file(f).exists
+    """Ensure that python2-specific files no longer exist, except on Debian 9."""
+    if host.system_info.distribution == "debian" and host.system_info.release == "9.12":
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]: Debian 9.13 was released on the 18th: https://www.debian.org/News/2020/20200718

**Review by mcdonnnj — APPROVED** [review]
Thank you for addressing this @jsf9k , strong work as usual 💪 I did have two items for your consideration, one of which might be breaking.

File: README.md (hunk 1 above)
Comment (mcdonnnj) [reply README.md]: Although now that I look at it again, if it's referring to the package name of "python2" I would prefer if that was more clear.

**Review by dav3r — APPROVED** [review]
LGTM, though please note @mcdonnnj's comment about Debian 9.13.  🐍

File: molecule/default/tests/test_default.py (hunk 2 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py]: Support for 9.13 added in commit e2adf63.

File: README.md (hunk 1 above)
Comment (jsf9k) [reply README.md]: Capitalized in commit 1adfd37.

**Review by dav3r — APPROVED** [review]
👍 👍

**Review by mcdonnnj — APPROVED** [review]
Thanks for taking care of those two comments, strong work as ever 💪

**Pull request merged**